duckduckgo/duckduckgo-autofill
# Autoprompt signin

**Reviewer:** @shakyShane 
**Asana:** https://app.asana.com/0/0/1202518354164389/f

## Description
- Adds autoprompt on both desktop and mobile. On mobile we also try to scroll the form into view and to submit the form.
- Adds testing.
- Minor improvements to detection accuracy.

## Steps to test
Added a bunch of automated tests, but here's how to test this on a real device:

- **On iOS/Android:**
  - Visit a page for which you have saved credentials, for example https://m.facebook.com/login.php.
  - If you're prompted by the cookie notice you should _not_ see an autoprompt.
    - In this case, you can dismiss the cookie notice and tap on the fields to bring up the autofill prompt as usual.
  - Otherwise you should immediately be prompted to login.
  - When selecting your credential, the form should be filled and submitted.

- **On macOS:**
  - Visit a page for which you have saved credentials, for example https://m.facebook.com/login.php.
  - If you're prompted by the cookie notice and it covers the form you should _not_ see an autoprompt.
    - In this case, you can dismiss the cookie notice and tap on the fields to bring up the autofill prompt as usual.
  - Otherwise you should find the tooltip open.
  - When selecting your credential, the form should be filled but _not_ submitted.
